localstack/services/cloudformation/service_models.py

```diff
@@ -551,14 +551,14 @@ def cloudformation_type():
         return 'AWS::StepFunctions::Activity'
 
     def fetch_state(self, stack_name, resources):
-        act_name = self.props.get('Name') or self.resource_id
-        act_name = self.resolve_refs_recursively(stack_name, act_name, resources)
-        sfn_client = aws_stack.connect_to_service('stepfunctions')
-        activities = sfn_client.list_activities()['activities']
-        result = [a['activityArn'] for a in activities if a['name'] == act_name]
-        if not result:
+        activity_arn = resources[self.resource_id].get('PhysicalResourceId')
```

Comment:
nitpick: we can better us this in the future:
```
activity_arn = self.physical_resource_id
```
... as this is returning `self.resource_json.get('PhysicalResourceId')` directly:
```
    @property
    def physical_resource_id(self):
        """ Return the (cached) physical resource ID. """
        return self.resource_json.get('PhysicalResourceId')
```